langkit/compiled_types.py

```diff
@@ -84,11 +84,21 @@ def template_extensions(ctx):
         'symbol_type':           T.Symbol.c_type(capi).name,
```

Comment:
(for the commit message): Could you add a high level description of this new mechanism? We also need some user-facing doc for this.